src/rdkafka_aux.c

```diff
@@ -332,3 +332,103 @@ uint16_t rd_kafka_Node_port(const rd_kafka_Node_t *node) {
 const char *rd_kafka_Node_rack(const rd_kafka_Node_t *node) {
         return node->rack;
 }
+
+/**
+ * @brief Creates a new rd_kafka_topic_partition_result_t object.
+ */
+
+rd_kafka_topic_partition_result_t *
+rd_kafka_topic_partition_result_new(const char *topic,
+                                    int32_t partition,
+                                    rd_kafka_resp_err_t err,
+                                    const char *errstr) {
+
+        rd_kafka_topic_partition_result_t *new_topic_partition;
+
+        new_topic_partition        = rd_calloc(1, sizeof(*new_topic_partition));
+        new_topic_partition->topic = rd_strdup(topic);
+        new_topic_partition->partition = partition;
+        new_topic_partition->err       = err;
+        new_topic_partition->errstr    = rd_strdup(errstr);
+
+        return new_topic_partition;
+}
+
+const char *rd_kafka_topic_partition_result_topic(
+    const rd_kafka_topic_partition_result_t *partition_result) {
+        return partition_result->topic;
+}
+
+int32_t rd_kafka_topic_partition_result_partition(
+    const rd_kafka_topic_partition_result_t *partition_result) {
+        return partition_result->partition;
+}
+
+rd_kafka_resp_err_t rd_kafka_topic_partition_result_error(
+    const rd_kafka_topic_partition_result_t *partition_result) {
+        return partition_result->err;
+}
+
+const char *rd_kafka_topic_partition_result_error_string(
+    const rd_kafka_topic_partition_result_t *partition_result) {
+        return partition_result->errstr;
+}
+
+/**
+ * @brief Returns the rd_kafka_topic_partition_result_t object at the given
+ *        index from the array of rd_kafka_topic_partition_result_t objects
+ *        from the rd_kafka_ElectLeaders_result_t event.
+ *
+ * @param result The rd_kafka_ElectLeadersResult_t object.
+ * @param idx The index of the rd_kafka_topic_partition_result_t object to
+ *
+ * @returns The rd_kafka_topic_partition_result_t object at the given index.
+ */
+const rd_kafka_topic_partition_result_t *rd_kafka_topic_partition_result_by_idx(
```

Comment:
1. Fix documentation for `rd_kafka_topic_partition_result_by_idx`. It is a generic class not specific to ElectLeaders.
2. Argument should be `partition_result` instead of `result`.

. Nothing to do if we are removing this function altogether.